testing/upload_validation.ts

```diff
@@ -1410,22 +1415,31 @@ function validateFormulas(schema: z.ZodObject<any>) {
             message:
               'Could not find a formula definition for this format. Each format must reference the name of a formula defined in this pack.',
           });
-        } else {
-          let hasError = !formula.parameters?.length;
-          const [_, ...extraParams] = formula.parameters || [];
-          for (const extraParam of extraParams) {
-            if (!extraParam.optional) {
-              hasError = true;
-            }
-          }
-          if (hasError) {
-            context.addIssue({
-              code: z.ZodIssueCode.custom,
-              path: ['formats', i],
-              message: 'Formats can only be implemented using formulas that take exactly one required parameter.',
-            });
+          return;
+        }
+        let hasError = !formula.parameters?.length;
+        const [_, ...extraParams] = formula.parameters || [];
+        for (const extraParam of extraParams) {
+          if (!extraParam.optional) {
+            hasError = true;
           }
         }
+        if (hasError) {
+          context.addIssue({
+            code: z.ZodIssueCode.custom,
+            path: ['formats', i],
+            message: 'Formats can only be implemented using formulas that take exactly one required parameter.',
+          });
+        }
+
+        if ((formula.matchers?.length || 0 > 0) && (format.matchers?.length || 0 > 0)) {
+          context.addIssue({
```

Comment:
I wonder if we should also block an isAction formula from having matchers?

Reply:
oh true good call